prowler/lib/outputs/compliance/compliance_output.py

```diff
@@ -0,0 +1,87 @@
+from abc import ABC, abstractmethod
+from io import TextIOWrapper
+from typing import List
+
+from prowler.lib.check.compliance_models import ComplianceBaseModel
+from prowler.lib.logger import logger
+from prowler.lib.outputs.finding import Finding
+from prowler.lib.utils.utils import open_file
+
+
+class ComplianceOutput(ABC):
```

Comment:
I think we can remove this class and change the compliance classes to inherit for Compliance, like:
```
class ComplianceOutput(Output):
    """
    This class represents an abstract base class for defining different types of outputs for findings.

    Attributes:
        _data (list): A list to store transformed data from findings.
        _file_descriptor (TextIOWrapper): A file descriptor to write data to a file.

    Methods:
        __init__: Initializes the Output class with findings, optionally creates a file descriptor.
        data: Property to access the transformed data.
        file_descriptor: Property to access the file descriptor.
        transform: Abstract method to transform findings into a specific format.
        batch_write_data_to_file: Abstract method to write data to a file in batches.
        create_file_descriptor: Method to create a file descriptor for writing data to a file.
    """

    def __init__(
        self,
        findings: list[Finding],
        compliance: ComplianceBaseModel
    ) -> None:
        self._data = []
        if findings:
            self.transform(findings)
            if create_file_descriptor:
                self.create_file_descriptor(file_path)
```

With that you can remove the rest of the file since the abstract methods are inherited.